Source/driver/Castro_generic_fill.cpp

```diff
@@ -6,89 +6,37 @@
 
 using namespace amrex;
 
-#ifdef __cplusplus
-extern "C"
+void ca_generic_fill(Box const& bx, FArrayBox& data,
+                     const int dcomp, const int numcomp,
+                     Geometry const& geom, const Real time,
+                     const Vector<BCRec>& bcr, const int bcomp,
+                     const int scomp)
 {
-#endif
-
-    // Note that these are called with dimension agnostic macros like
-    // AMREX_ZFILL and AMREX_INT_ANYD already, so we should expect that
-    // everything has three entries, not AMREX_SPACEDIM entries.
-    // We still choose to use the macros anyway below, for compatibility
-    // with the GPU pragma script.
+    // Make a copy of the raw BCRec data in the format
+    // our BC routines can handle (a contiguous array
+    // of integers).
```

Comment:
will this be undone when we move the fills to C++?

Reply:
The AMReX calls in C++ can directly take the Vector<BCRec> without any preprocessing on our parts. For the C++ calls that we implement, we won't need to serialize the vector like this.